src/mon/MDSMonitor.cc

```diff
@@ -203,7 +203,7 @@ void MDSMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     }
     for (const auto &metric : health.metrics) {
       int const rank = info.rank;
-      health_check_t *check = &new_checks.add(
+      health_check_t *check = &new_checks.get_or_add(
```

Comment:
I think with Sage's change we are still accumulating the information about each unhealthy MDS in the `detail` part of the check.  The message in the `summary` part is overwritten, but it would always be the same because it is just determined from the health check code.  That message has a %num in it, so when two MDSs raise the same health metric, we'll get e.g. "2 MDSs behind on trimming" as expected.

We do need to do this across all MDSs, because the checks are daemon-local things, that are not necessarily the same across the cluster.  e.g. it is quite possible for one MDS to be behind on trimming its journal but the others not to be.